ui/src/dashboards/components/NoteEditorOverlay.tsx

```diff
@@ -92,34 +87,56 @@ class NoteEditorOverlay extends PureComponent<Props, State> {
   }
 
   public render() {
+    const {dashboardID} = this.props
+
+    if (!dashboardID) {
+      return (
+        <Overlay.Container maxWidth={360}>
+          <Overlay.Header title="Oh no!" onDismiss={this.handleDismiss} />
+          <Overlay.Body>
+            <h5>
+              This page does not allow creation or editing of notes, better head{' '}
+              to a dashboard to do that.
```

Comment:
Do you know under what circumstances a user will see error message? Is this a protection against engineers putting overlays in the wrong place after we move forward with the new system?

Reply:
Now that you mention it, it is a vestige of this overlay being displayed potentially everywhere. Now that it is only accessible via a single route I don't think it's an issue. I did want to guard against the lack of params since it needs them to render correctly, but maybe there's a more generic & reusable way to handle it